src/utils/MessageBuilder.ts

```diff
@@ -0,0 +1,119 @@
+import { MessageActionRow, MessageButton, MessageEmbed, MessageOptions } from "discord.js";
+import { PlayerInQueue } from "../repositories/QueueRepository/types";
+
+export const enum ButtonCustomID {
+  JoinQueue = "joinQueue",
+  LeaveQueue = "leaveQueue",
+}
+
+export default class MessageBuilder {
+  private static readonly normIconURL =
+    "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png";
+
+  static leaderboardMessage(leaderboardInfo: string[]): MessageOptions {
+    const embeds = leaderboardInfo.map((content, index) => {
+      const embedCtr = leaderboardInfo.length > 1 ? `(${index + 1}/${leaderboardInfo.length})` : "";
+
+      return new MessageEmbed({
+        color: "BLUE",
+        description: "```" + content + "```",
+        thumbnail: { url: this.normIconURL },
+        title: `UNCC 6 Mans | Leaderboard ${embedCtr}`.trim(),
+      });
+    });
+
+    return {
+      embeds,
+    };
+  }
+
+  static disabledButtonsJoining(): MessageOptions {
```

Comment:
Instead of having two different versions of disabled, would it make sense to just condense this to one `disabledQueueButtons` where both buttons disable and say "Please wait..."?

Reply:
Resolved.